tools/mage/git.go

```diff
@@ -199,9 +196,64 @@ nextTopic:
 	return nil
 }
 
-// RunHook runs the Git hook for $HOOK, arguments are taken from $ARGS.
+func (g Git) prePush(stdin string, args ...string) error {
+	if mg.Verbose() {
+		fmt.Println("Running pre-push hook")
+	}
+	if stdin == "" {
+		return errors.New("standard input is empty")
+	}
+	var (
+		ref  string
+		head string
+	)
+	if ss := strings.Fields(stdin); len(ss) == 4 {
+		ref = ss[0]
+		head = ss[1]
+	} else {
+		return fmt.Errorf("expected pre-push hook standard input to contain 4 fields, got: %d(`%s`)", len(ss), ss)
+	}
+	if len(args) != 2 {
+		return fmt.Errorf("pre-push hook expected to get 2 arguments, got: %s", args)
+	}
+	const ttesMarkerHash = "a184c9c64d31580ecfc71ce55f1504d191e185e1"
+	switch err := exec.Command("git", "merge-base", "--is-ancestor", ttesMarkerHash, head).Run().(type) {
```

Comment:
`sh.Run` makes all kind of weirdness with the command, it makes no sense to use it here - it's more verbose and requires workarounds to check the exit code